contracts/src/MultiAddress.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+pragma solidity 0.8.22;
+
+using {isIndex, asIndex, isAddress32, asAddress32, isAddress20, asAddress20} for MultiAddress global;
+
+/// @dev An address for an on-chain account
+struct MultiAddress {
+    Kind kind;
+    bytes data;
+}
+
+enum Kind {
+    Index,
```

Comment:
What is this `Index` kind? A parachain ID?

Reply:
If you install [pallet_indices](https://paritytech.github.io/polkadot-sdk/master/pallet_indices/index.html) on a substrate chain, then its possible for the chain to map a `u32` to a `AccountId` under the hood.

So its basically a "short" id for accountid.

In fact Substrate allows all sorts of mappings:
https://paritytech.github.io/polkadot-sdk/master/sp_runtime/enum.MultiAddress.html